ca/ca.go

```diff
@@ -277,7 +280,11 @@ func (ca *CAImpl) newCertificate(domains []string, ips []net.IP, key crypto.Publ
 		}
 	}
 
-	certNotAfter := time.Now().AddDate(5, 0, 0)
+	certNotAfter := time.Now().AddDate(0, 0, ca.certLifetime)
```

Comment:
Once `ca.certValidityPeriod` is measured in seconds, this can become `.Add(ca.certValidityPeriod * time.Second)`.

Also, this is unrelated to the change you're making now, but it'd be great if you did it while you're here: subtract one second, to account for the way validity periods are actually computed. So really this should be `.Add(ca.certValidityPeriod * time.Second).Add(-1 * time.Second)` (or of course you could combine them into a single operation if you think that's still clear).

Reply:
Had to use `time.Duration()` in combination with `time.Second()` otherwise Go would complain about `uint` not being type `time.Duration`...